src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -29,6 +38,10 @@
     @Autowired
     private OpenLibraryService openLibraryService;
 
+    private static final String[] array = {"genre", "author", "title", "image", "subtitle", "publisher", "year", "ages", "id"};
+    private static final LinkedList permittedSort = new LinkedList(Arrays.asList(array));
```

Comment:
Maybe you could simplify it by doing something like this: 

`private static final List permittedSort = Arrays.asList("genre", "author", ...)`

that way you don't need two different fields for the same thing. Also, while LinkedList can be more efficient for some operations, in this case it doesn't make a big difference. There are limited elements that you'll only add once and the only operation you will perform is `.contains`, which has linear complexity either way. And since the number of elements is constant we can even consider the complexity as O(1) :) 

Same goes for the UserController class